project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module

```diff
@@ -170,113 +525,1600 @@ function c4m_message_load__c4m_welcome_to_group(Message $message) {
+    'pass message' => TRUE,
+  );
+
+  $message->arguments['@event_user_full_name'] = array(
+    'callback' => 'c4m_message_format_event_user_full_name',
+    'pass message' => TRUE,
+  );
+
+  $message->arguments['!help_section_url'] = array(
+    'callback' => 'c4m_message_format_help_section_url',
+    'pass message' => TRUE,
+  );
+}
+
+/**
+ * Returns message addressee user ID.
+ *
+ * @param Message $message
+ *   The Message entity - used to retrieve required info.
+ *
+ * @return int
+ *   Returns addressee's user ID.
+ */
+function c4m_message_format_addressee_uid(Message $message) {
+  return $message->uid;
+}
+
+/**
+ * Returns the of group message refers to.
+ *
+ * @param Message $message
+ *   The Message entity.
+ *
+ * @return int
+ *   Returns group ID.
+ */
+function c4m_message_format_group_gid(Message $message) {
+  $wrapper = entity_metadata_wrapper('message', $message);
+
+  return $wrapper->field_group_node->value(array('identifier' => TRUE));
+}
+
+/**
+ * Returns addressee full name token.
+ *
+ * @param Message $message
+ *   The Message entity.
+ *
+ * @return string
+ *   Returns addressee's full name.
+ */
+function c4m_message_format_addressee_full_name(Message $message) {
+  $user = user_load($message->uid);
+
+  return c4m_user_name($user);
+}
+
+/**
+ * Returns new member full name token.
+ *
+ * @param Message $message
+ *   The Message entity.
+ *
+ * @return string
+ *   Returns member's full name.
+ */
+function c4m_message_format_new_member_full_name(Message $message) {
+  $wrapper = entity_metadata_wrapper('message', $message);
+  $new_member = $wrapper->field_related_user->value();
+
+  return c4m_user_name($new_member);
+}
+
+/**
+ * Returns new member organisation.
+ *
+ * @param Message $message
+ *   The Message entity.
+ *
+ * @return string
+ *   Returns member's organisation.
+ */
+function c4m_message_format_new_member_organisation(Message $message) {
+  $wrapper = entity_metadata_wrapper('message', $message);
+
+  return $wrapper->field_related_user->c4m_organisation->value();
+}
+
+/**
+ * Returns new member organisation type.
+ *
+ * @param Message $message
+ *   The Message entity.
+ *
+ * @return string
+ *   Returns member's organisation type.
+ */
+function c4m_message_format_new_member_organisation_type(Message $message) {
+  $wrapper = entity_metadata_wrapper('message', $message);
+
+  return $wrapper->field_related_user->c4m_organisation_type->value();
+}
+
+/**
+ * Returns new member country.
+ *
+ * @param Message $message
+ *   The Message entity.
+ *
+ * @return string
+ *   Returns member's country.
+ */
+function c4m_message_format_new_member_country(Message $message) {
+  $wrapper = entity_metadata_wrapper('message', $message);
+  $field = field_info_field('c4m_country');
+  $allowed_values = list_allowed_values($field);
+  $index = $wrapper->field_related_user->c4m_country->value();
+
+  return empty($index) ? '' : $allowed_values[$index];
```

Comment:
`c4m_country` can be empty for migrated users as this field was not mandatory in the old platform.

Regarding user 1:
Only if he legitimately should receive them (not because he is user "1"): so only if he is group owner, manager...